src/components/workspace/editorMode/source.js

```diff
@@ -0,0 +1,831 @@
+        };
+    
+    }).call(MatchingBraceOutdent.prototype);
+    
+    exports.MatchingBraceOutdent = MatchingBraceOutdent;
+    });
+    
+    ace.define("ace/mode/folding/cstyle",["require","exports","module","ace/lib/oop","ace/range","ace/mode/folding/fold_mode"], function(acequire, exports, module) {
+    "use strict";
+    
+    var oop = acequire("../../lib/oop");
+    var Range = acequire("../../range").Range;
+    var BaseFoldMode = acequire("./fold_mode").FoldMode;
+    
+    var FoldMode = exports.FoldMode = function(commentRegex) {
+        if (commentRegex) {
+            this.foldingStartMarker = new RegExp(
+                this.foldingStartMarker.source.replace(/\|[^|]*?$/, "|" + commentRegex.start)
+            );
+            this.foldingStopMarker = new RegExp(
+                this.foldingStopMarker.source.replace(/\|[^|]*?$/, "|" + commentRegex.end)
+            );
+        }
+    };
+    oop.inherits(FoldMode, BaseFoldMode);
+    
+    (function() {
+        
+        this.foldingStartMarker = /([\{\[\(])[^\}\]\)]*$|^\s*(\/\*)/;
+        this.foldingStopMarker = /^[^\[\{\(]*([\}\]\)])|^[\s\*]*(\*\/)/;
+        this.singleLineBlockCommentRe= /^\s*(\/\*).*\*\/\s*$/;
+        this.tripleStarBlockCommentRe = /^\s*(\/\*\*\*).*\*\/\s*$/;
+        this.startRegionRe = /^\s*(\/\*|\/\/)#?region\b/;
+        this._getFoldWidgetBase = this.getFoldWidget;
+        this.getFoldWidget = function(session, foldStyle, row) {
+            var line = session.getLine(row);
+        
+            if (this.singleLineBlockCommentRe.test(line)) {
+                if (!this.startRegionRe.test(line) && !this.tripleStarBlockCommentRe.test(line)) {
+                    return "";
+                }
+            }
+        
+            var fw = this._getFoldWidgetBase(session, foldStyle, row);
+        
+            if (!fw && this.startRegionRe.test(line)) {
+                return "start";
+            } // lineCommentRegionStart
+        
+            return fw;
+        };
+    
+        this.getFoldWidgetRange = function(session, foldStyle, row, forceMultiline) {
+            var line = session.getLine(row);
+            var i = 0;
+            
+            if (this.startRegionRe.test(line)) {
+                return this.getCommentRegionBlock(session, line, row);
+            }
+            
+            var match = line.match(this.foldingStartMarker);
+            if (match) {
+                i = match.index;
+    
+                if (match[1]) {
+                    return this.openingBracketBlock(session, match[1], row, i);
+                }
+                    
+                var range = session.getCommentFoldRange(row, i + match[0].length, 1);
+                
+                if (range && !range.isMultiLine()) {
+                    if (forceMultiline) {
+                        range = this.getSectionRange(session, row);
+                    } else if (foldStyle !== "all") {
+                        range = null;
+ }
```

Comment:
Looks like this bracket is at the wrong indent level!

Reply:
deleted ace.define function for Folding/CStyle as well since I realised that it should be irrelevant.